lib/promises.js

```diff
@@ -0,0 +1,61 @@
+'use strict';
+
+/// encapsulate a method with a node-style callback in a Promise
+/// @param {object} 'this' of the encapsulated function
+/// @param {function} function to be encapsulated
+/// @param {Array-like} args to be passed to the called function
+/// @return {Promise} a Promise encapuslaing the function
+module.exports.promise = function (fn, context, args) {
+
+    //polyfill (for 0.12)
+    //avoid [].slice.call due to performance issues on v8
```

Comment:
The “performance problem” already exists when you pass `arguments` to `exports.promise`, and exists as much for `Array.from` as it exists for `[].slice.call`. Since the JavaScript part of the bcrypt bindings isn’t the slow one, though, it doesn’t really matter and you don’t need to comment on it here.